src/pow.cpp

```diff
@@ -354,13 +354,14 @@ void multinomial_coefficients_mpz(int m, int n, map_vec_mpz &r)
 
 RCP<const Basic> pow_expand(const RCP<const Pow> &self)
 {
-    if (! is_a<Integer>(*self->exp_) || ! is_a<Add>(*self->base_))
+    RCP<const Basic> _base = expand(self->base_);
```

Comment:
This is where the benchmarks are getting slower. I will take a look at them to optimize it. This expand is necessary in case base_ has a `Mul` or a `Pow` that needs to be expanded.